pkg/v1/tkg/test/tkgctl/shared/package.go

```diff
@@ -158,27 +162,33 @@ func getPackagesFromCB(ctx context.Context, clusterBootstrap *runtanzuv1alpha3.C
 			if pkgShortName == "tkg-storageclass" {
 				continue
 			}
-			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: constants.TkgNamespace},
+			packages = append(packages, kapppkgv1alpha1.Package{ObjectMeta: metav1.ObjectMeta{Name: pkgShortName, Namespace: systemNamespace},
 				Spec: kapppkgv1alpha1.PackageSpec{RefName: pkgName, Version: pkgVersion}})
 		}
 	}
 
 	return packages, nil
 }
 
-// checkClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
-func checkClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
+// CheckClusterCB checks if clusterbootstrap resource is created correctly and packages are reconciled successfully on cluster
+func CheckClusterCB(ctx context.Context, mccl, wccl client.Client, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName string, isManagementCluster bool) error {
 	log.Infof("Verify addons on workload cluster %s with management cluster %s", wcClusterName, mcClusterName)
 
 	var clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap
+
+	systemNamespace := constants.TkgNamespace
+	if infrastructureName == "tkgs" {
+		systemNamespace = constants.TKGSClusterClassNamespace
+	}
+
 	if isManagementCluster {
 		clusterBootstrap = getClusterBootstrap(ctx, mccl, mcClusterNamespace, mcClusterName)
 	} else {
 		clusterBootstrap = getClusterBootstrap(ctx, mccl, wcClusterNamespace, wcClusterName)
 	}
 
 	By(fmt.Sprintf("Verify clusterbootstrap matches clusterbootstraptemplate"))
-	verifyClusterBootstrap(ctx, mccl, clusterBootstrap, clusterBootstrap.Status.ResolvedTKR)
+	verifyClusterBootstrap(ctx, mccl, clusterBootstrap, clusterBootstrap.Status.ResolvedTKR, systemNamespace)
 
 	packages, err := getPackagesFromCB(ctx, clusterBootstrap, mccl, wccl, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName, isManagementCluster)
 	Expect(err).NotTo(HaveOccurred())
```

Comment:
```suggestion
if clusterBootstrap == nil {
     return
}

By(fmt.Sprintf("Verify clusterbootstrap matches clusterbootstraptemplate"))
verifyClusterBootstrap(ctx, mccl, clusterBootstrap, clusterBootstrap.Status.ResolvedTKR, systemNamespace)
packages, err := getPackagesFromCB(ctx, clusterBootstrap, mccl, wccl, mcClusterName, mcClusterNamespace, wcClusterName, wcClusterNamespace, infrastructureName, isManagementCluster)
Expect(err).NotTo(HaveOccurred())

Reply:
The reason for the above suggestion is that for supervisor cluster specifically, there is no CB, and the following checks would panic for that case

Reply:
Instead of checking for the nilness of ```clusterBootstrap```, alternatively you can do the following check. You decide which one you prefer:
```
  if isManagementCluster && infrastructureName == "tkgs" {
        return nil 
 }
```

Reply:
As mentioned before, for tkgs tests we are not calling this for supervisor cluster.

Reply:
Oh ok, we are good then